src/carplay.cs

```diff
@@ -0,0 +1,719 @@
+
+		[Abstract]
+		[Export ("searchTemplate:selectedResult:completionHandler:")]
+		void SelectedResult (CPSearchTemplate searchTemplate, CPListItem item, Action completionHandler);
+
+		[Export ("searchTemplateSearchButtonPressed:")]
+		void SearchButtonPressed (CPSearchTemplate searchTemplate);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPSessionConfiguration {
+
+		[Export ("initWithDelegate:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (ICPSessionConfigurationDelegate @delegate);
+
+		[Export ("limitedUserInterfaces")]
+		CPLimitableUserInterface LimitedUserInterfaces { get; }
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		ICPSessionConfigurationDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+	}
+
+	interface ICPSessionConfigurationDelegate { }
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[Protocol, Model ( AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface CPSessionConfigurationDelegate {
+
+		[Abstract]
+		[Export ("sessionConfiguration:limitedUserInterfacesChanged:")]
+		void LimitedUserInterfacesChanged (CPSessionConfiguration sessionConfiguration, CPLimitableUserInterface limitedUserInterfaces);
+	}
+
+	[Abstract]
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	interface CPTemplate : NSSecureCoding {
+
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPRouteChoice : NSCopying, NSSecureCoding {
+
+		[Export ("initWithSummaryVariants:additionalInformationVariants:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string [] summaryVariants, string [] additionalInformationVariants);
+
+		[Export ("summaryVariants", ArgumentSemantic.Copy)]
+		string [] SummaryVariants { get; }
+
+		[NullAllowed, Export ("additionalInformationVariants", ArgumentSemantic.Copy)]
+		string [] AdditionalInformationVariants { get; }
+
+		[NullAllowed, Export ("userInfo", ArgumentSemantic.Strong)]
+		NSObject UserInfo { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CPTrip : NSSecureCoding {
+
+		[Export ("initWithOrigin:destination:routeChoices:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MKMapItem origin, MKMapItem destination, CPRouteChoice [] routeChoices);
+
+		[Export ("origin", ArgumentSemantic.Strong)]
+		MKMapItem Origin { get; }
+
+		[Export ("destination", ArgumentSemantic.Strong)]
+		MKMapItem Destination { get; }
+
+		[Export ("routeChoices", ArgumentSemantic.Copy)]
+		CPRouteChoice [] RouteChoices { get; }
+
+		[NullAllowed, Export ("userInfo", ArgumentSemantic.Strong)]
+		NSObject UserInfo { get; set; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (12,0)]
+	[BaseType (typeof (CPTemplate))]
+	interface CPVoiceControlTemplate {
```

Comment:
still in such case I don't think the default ctor should be included because the `init*` call does not accept null parameters

Reply:
ahh dang, I thought I caught them all :pokeball: nice catch!